tests/unit/test_type_hinting.py

```diff
@@ -95,15 +96,37 @@ def test_hint_comparisons(self):
                 typing.Callable[[int, float], float],
                 False,
             ),
+            (
+                typing.Annotated[int, "foo"],
+                int,
+                True,
+            ),
+            (
+                int,
+                typing.Annotated[int, "foo"],
+                True,
+            ),
         ]:
             with self.subTest(
                 target=target, reference=reference, expected=is_more_specific
             ):
                 self.assertEqual(
                     type_hint_is_as_or_more_specific_than(target, reference),
                     is_more_specific,
+                    msg=f"{target} is {'not ' if not is_more_specific else ''}more specific than {reference}",
                 )
 
+    def test_get_type_hints(self):
+        for hint, origin in [
+            (int | float, type(int| float)),
+            (typing.Annotated[int | float, "foo"], type(int | float)),
+            (int, None),
+            (typing.Annotated[int, "foo"], None),
```

Comment:
Could we extend this to include hints that specify generics? E.g.

```suggestion
            (typing.Annotated[int, "foo"], None),
            (typing.Annotated[list[int], list),
            (list[int], list),
```

Reply:
I edited it because there was a typo, and `typing.Annotated` doesn't work without metadata.